File: eel-core/src/main/scala/io/eels/HdfsOps.scala
package io.eels

import com.sksamuel.exts.Logging
import io.eels.util.{HdfsIterator, PathIterator}
import org.apache.hadoop.fs.{FileSystem, LocatedFileStatus, Path}

object HdfsOps extends Logging {

  def makePathVisible(path: Path)(implicit fs: FileSystem): Unit = {
    if (path.getName.startsWith(".")) {
      logger.info(s"Making $path visible by stripping leading .")
      val dest = new Path(path.getParent, path.getName.drop(1))
      fs.rename(path, dest)
    }
  }

  def findFiles(path: Path, recursive: Boolean, fs: FileSystem): Iterator[LocatedFileStatus] = {
    HdfsIterator(fs.listFiles(path, recursive))
  }

  def mkdirsp(path: Path, fs: FileSystem): Boolean = PathIterator(path).forall(fs.mkdirs)
}

File: eel-core/src/main/kotlin/io/eels/HdfsOps.kt
package io.eels

import org.apache.hadoop.fs.FileSystem
import org.apache.hadoop.fs.LocatedFileStatus
import org.apache.hadoop.fs.Path

object HdfsOps {

  fun findFiles(path: Path, recursive: Boolean, fs: FileSystem): Iterator<LocatedFileStatus> {
    return HdfsIterator(fs.listFiles(path, recursive))
  }

  fun mkdirsp(path: Path, fs: FileSystem): Boolean = PathIterator(path).asSequence().all { fs.mkdirs(it) }
}